tensorboard/webapp/plugins/npmi/store/npmi_reducers.ts

```diff
@@ -80,6 +104,250 @@ const reducer = createReducer(
         },
       };
     }
+  ),
+  on(
+    actions.npmiAddSelectedAnnotations,
+    (state: NpmiState, {annotations}): NpmiState => {
+      return {
+        ...state,
+        selectedAnnotations: [
+          ...new Set([...state.selectedAnnotations, ...annotations]),
+        ],
+      };
+    }
+  ),
+  on(
+    actions.npmiRemoveSelectedAnnotation,
+    (state: NpmiState, {annotation}): NpmiState => {
+      const annotationSet = new Set([...state.selectedAnnotations]);
+      annotationSet.delete(annotation);
+      return {
+        ...state,
+        selectedAnnotations: [...annotationSet],
+      };
+    }
+  ),
+  on(
+    actions.npmiSetSelectedAnnotations,
+    (state: NpmiState, {annotations}): NpmiState => {
+      return {
+        ...state,
+        selectedAnnotations: annotations,
+      };
+    }
+  ),
+  on(
+    actions.npmiClearSelectedAnnotations,
+    (state: NpmiState): NpmiState => {
+      return {
+        ...state,
+        selectedAnnotations: [],
+      };
+    }
+  ),
+  on(
+    actions.npmiToggleAnnotationFlags,
+    (state: NpmiState, {annotations}): NpmiState => {
+      const combinedFlaggedAnnotations = new Set([
+        ...state.flaggedAnnotations,
+        ...annotations,
+      ]);
+      if (combinedFlaggedAnnotations.size === state.flaggedAnnotations.length) {
+        // If all annotations are already flagged, user wants to remove them
+        for (const annotation of annotations) {
+          combinedFlaggedAnnotations.delete(annotation);
+        }
+      }
+      return {
+        ...state,
+        flaggedAnnotations: [...combinedFlaggedAnnotations],
+      };
+    }
+  ),
+  on(
+    actions.npmiToggleAnnotationsHidden,
+    (state: NpmiState, {annotations}): NpmiState => {
+      const notHidden: string[] = annotations.filter(
```

Comment:
Could you take a similar approach as `npmiToggleAnnotationFlags` here?

Reply:
Sure, sorry I missed that.